src/veh_interact.cpp

```diff
@@ -316,30 +315,19 @@ task_reason veh_interact::cant_do (char mode)
         } );
         has_tools = true; // checked later
         break;
-    case 'f': // refill mode
-        if (!ptanks.empty()) {
-            std::vector<vehicle_part *>::iterator iter;
-            for (iter = ptanks.begin(); iter != ptanks.end(); ) {
-                if ((*iter)->hp > 0 &&
-                    g->refill_vehicle_part(*veh, *iter, true)) {
-                    pass_checks = true;
-                    iter++;
-                } else {
-                    iter = ptanks.erase(iter);
-                }
-            }
-            if(pass_checks) {
-                return CAN_DO;
-            }
-        }
 
-        // if refillable parts exist but can't be refilled for some reason.
-        if (has_ptank) {
-            return CANT_REFILL;
-        }
+    case 'f': {
+        // first find all possible ammo
+        auto opts = g->u.items_with( []( const item &e ) { return e.is_ammo(); } );
```

Comment:
> How do you propose to stop such support bitrotting? Emphasis mine but in the lack of any actual in-the-wild use examples I think that's a fair assessment.

We don't need to. It works fine. If you're worried, add a new vehicle tank part that uses for example salt water as fuel type and the feature is not unused anymore.

I have not seen any good reason why the fuel type *needs* the restriction. Especially as it encourages other hacks (the mentioned addition of ammo data to allow storing bleach in vehicle tanks).

> If we allow non-ammo what should  ammo_current()  and  ammo_type()  return for non-ammo fuels?

> I want to avoid the special case where  ammo_current() == "null"  but  ammo_remaining() > 0  if at all possible. The exists in the  item  code due to legacy items (spray can, lighter etc) but has been the cause of a lot of bugs and adds a lot of boilerplate to the code.

Why not add a `content_count(ammotype)` function (different name maybe), which returns the charges of the content (if the content is ammo and has a matching ammo type) or 0 otherwise.

And with `ammotype` being a separate type, you can overload the function to take a `itype_id` to check how much of a given item it contains.

And maybe a third overload without parameters to get the content charges of any content type.

--

Regarding items like spray can or lighter: we probably have to decide - do they contain ammo or do they only contain "charges" (an arbitrary number of usages)?

If ammo, they should get a proper ammo type (and flags to prevent unloading/reloading as needed). Cash cards for example use this.

If "charges", add a `get_charges` function (or `get_remaining_usages`) that returns exactly that. Similar to `get_remaining_chapters` for books.